### PR TITLE
Set up new gh ratelimit publisher for gateway/temporalworker

### PR DESCRIPTION
**Pull request by samrabelachew**
Two changes here:
1. Inject installation ID defining a separate GH app for gateway and temporal servers (we will [eventually](https://github.com/lyft/atlantis/pull/711) reconfigure code to use these installation ids over the ones passed in from the webhook event)
2. Set up new github ratelimit stats publisher in gateway and temporal servers that use these app installations. This already needed to be migrated over to the new workers anyways, but also can serve as an initial test of the new ids without having to impact existing code.

### CODE REVIEW

File: server/neptune/gateway/event/push_event_handler_test.go [hunk 1]
```diff
@@ -8,29 +8,13 @@ import (
...
 	"github.com/runatlantis/atlantis/server/neptune/gateway/event"
-	"github.com/runatlantis/atlantis/server/neptune/lyft/feature"
 	"github.com/runatlantis/atlantis/server/neptune/sync"
 	"github.com/runatlantis/atlantis/server/vcs"
 	"github.com/stretchr/testify/assert"
 )
 
 const testRoot = "testroot"
 
-type testAllocator struct {
```
Comment (samrabelachew) [inline server/neptune/gateway/event/push_event_handler_test.go:19]: Deleting this is an unrelated lint fix (not used code)

File: server/neptune/sync/crons/ratelimit_stats.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		ClientCreator:  clientCreator,
+		InstallationID: installationID,
+	}
+}
+
+func (r *RateLimitStatCollector) Run(ctx context.Context) error {
+	installationClient, err := r.ClientCreator.NewInstallationClient(r.InstallationID)
+	if err != nil {
+		return errors.Wrap(err, "creating installation client")
+	}
```
Comment (mikecutalo) [inline server/neptune/sync/crons/ratelimit_stats.go:35]: is this creating a new client every time? or does it pass back one thats already been created?

Comment (samrabelachew) [reply server/neptune/sync/crons/ratelimit_stats.go:35]: ClientCreator comment:
```
// NewCachingClientCreator returns a ClientCreator that creates a GitHub client for installations of the app specified
// by the provided arguments. It uses an LRU cache of the provided capacity to store clients created for installations
// and returns cached clients when a cache hit exists.
```

See code itself for cache layer: https://github.com/palantir/go-githubapp/blob/develop/githubapp/caching_client_creator.go#L74

**Review by mikecutalo — APPROVED** [review]
